clang/lib/3C/ConstraintResolver.cpp

```diff
@@ -249,14 +227,37 @@ CVarSet ConstraintResolver::getExprConstraintVars(Expr *E) {
     if (CHKCBindTemporaryExpr *CE = dyn_cast<CHKCBindTemporaryExpr>(E)) {
       return getExprConstraintVars(CE->getSubExpr());
     }
+
     // Apart from the above expressions constraints for all the other
     // expressions can be cached.
     // First, check if the expression has constraints that are cached?
     if (Info.hasPersistentConstraints(E, Context))
       return Info.getPersistentConstraints(E, Context);
 
     CVarSet Ret = EmptyCSet;
-    if (ExplicitCastExpr *ECE = dyn_cast<ExplicitCastExpr>(E)) {
+    // Implicit cast, e.g., T* from T[] or int (*)(int) from int (int),
+    //   but also weird int->int * conversions (and back)
+    if (ImplicitCastExpr *IE = dyn_cast<ImplicitCastExpr>(E)) {
```

Comment:
Why did this get moved to another spot in the `if` / `then` / `else` ? IIRC, the order of these things was relevant, but I might be wrong about that.

Reply:
And I'm not seeing how it's different from the original code that was moved?

Reply:
This is moved because order is important. It was originally in the `if` branch so that the implicit cast was not stored in the constraint variables map. Now that there is a map specifically for implicit casts, the code needs to be in the other block.